axelrod/match.py

```diff
@@ -77,11 +82,28 @@ def play(self):
 
         i.e. One entry per turn containing a pair of actions.
         """
+        if self._prob_end is not None:
```

Comment:
Ok, you've obviously thought about this quite a bit, so I'm fine with this approach.